src/VoteView.js

```diff
@@ -2,25 +2,51 @@ import React, { useEffect, useState } from 'react';
 import styled from 'styled-components';
 import axios from 'axios';
 import CandidateVotes from './Candidate';
+import { Link } from 'react-router-dom';
+import { Button } from './Candidate';
+
 const url =
   'http://ec2-13-209-5-166.ap-northeast-2.compute.amazonaws.com:8000/api/candidates';
 
-function VoteView() {
+function LoginoutButton({ isLogined, handleLogoutButton }) {
+  if (isLogined) {
+    return (
+      <UserButton onClick={handleLogoutButton} r={128} g={203} b={196}>
+        로그아웃
+      </UserButton>
+    );
+  } else
+    return (
+      <Link to="/signin">
+        <UserButton r={128} g={203} b={196}>
+          로그인
+        </UserButton>
+      </Link>
+    );
+}
+
+function VoteView({ loginCookie, removeLoginCookie }) {
   const [candidates, setCandidates] = useState([]);
   const [voteFlag, setVoteFlag] = useState(false);
+  const [isLogined, setIsLogined] = useState(Boolean(loginCookie.loginCookie));
```

Comment:
```suggestion
  const [isLoggedIn, setIsLoggedIn] = useState(!!(loginCookie.loginCookie));
```

boolean으로 바꾸는 데 `!!`을 쓰면 더 간결해진다. 꿀팁!

그리고  `isLoggedIn`이 여기에 state로 있으면 안되지!
App.js 최상위 컴포넌트에 있어야 코드를 효과적으로 쓸 수 있어